auacm/app/modules/submission_manager/judge.py

```diff
@@ -42,182 +46,169 @@
+            shlex.split(COMPILE_COMMAND[self.submission.file_type]
+                        .format(os.path.join(directory, name))),
+            stderr=open(os.path.join(directory, 'error.txt'), 'w')
+        )
+
+        if result == 0:
+            return COMPILATION_SUCCESS
+        else:
+            return COMPILATION_ERROR
+
+
+    def execute_submission(self):
+        """Run the submission.
+
+        This method:
+            1. detects all the input files associated with this problem.
+            2. runs the submission with each input file
+            3. checks the performance of the submission for errors (TLE, RTE, etc)
+            4. compares the output against correct test output
+        """
+        # Initial setup
+        input_path = os.path.join(self.directory_for_problem, 'in')
+        output_path = os.path.join(self.directory_for_submission, 'out')
+        max_runtime = self.time_limit * TIMEOUT_MULTIPLIER[self.submission.file_type]
+
         # Final setup.
-        directory = directory_for_submission(submit)
-        output_path = os.path.join(directory, 'out')
-        if (not os.path.exists(output_path)):
+        output_path = os.path.join(self.directory_for_submission, 'out')
+        if not os.path.exists(output_path):
             os.mkdir(output_path)
 
-    def run(self):
-        """Execute a subprocess and keep the pointer to that subprocess."""
-        start_time = time.time()
-        self.process = self.judge_as_subprocess()
-        # TODO(djshuckerow): Get this thread to join() properly.
-        while self.process.poll() is None:
-            time.sleep(0.1)
-            if time.time() > start_time + self.limit:
-                # The try is to avoid a race condition where the process
-                # finishes between the if and the kill statements.
+        # Iterate over all the input files.
+        for fname in os.listdir(input_path):
+            f = os.path.join(input_path, fname)
+            if os.path.isfile(f):
+                # Prepare to run the test file.
+                test_number = int(fname.split('.')[0].strip('in'))
+                out_file = 'out{0}.txt'.format(test_number)
+
+                process = self.create_process(f, out_file)
                 try:
-                    self.process.kill()
-                except:
-                    pass
+                    process.communicate(timeout=max_runtime)
+                except subprocess.TimeoutExpired:
+                    process.kill()
+                    return TIMELIMIT_EXCEEDED, test_number
+
+                if process.poll() != 0:
+                    return RUNTIME_ERROR, test_number
+
+                result_path = os.path.join(self.directory_for_problem, 'out')
+
+                # The execution is completed.  Check its correctness.
+                with open(os.path.join(output_path, out_file)) as correct_result, \
+                     open(os.path.join(result_path, out_file)) as submission_result:
```

Comment:
Line continuation with [parentheses](https://google.github.io/styleguide/pyguide.html?showone=Line_length#Line_length) is prefered over backslash.